### PR TITLE
i345 - Update IIIF Manifest labels to show titles

### PR DESCRIPTION
**Pull request by kirkkwang**
ref #345 

Prior to this commit, the IIIF Manifest's canvas label is based on the record's label which is actually the filename.  This is not desirable because the filename is often times not human-readable.  This commit will instead instead the file set's parent work's title and use that instead as the label.

Before:
image

After:
image

The new name is `The Mysticism of Ignatius - Page 01` and so on.

### CODE REVIEW

**Comment by laritakr** [comment]
This is a bit misleading with the changes to the way we do child work titles.

Updated examples coming up...

**Comment by kirkkwang** [comment]
> This is a bit misleading with the changes to the way we do child work titles.
> 
> Updated examples coming up...

oh nooo you're right, we wouldn't have such nice titles

**Comment by ShanaLMoore** [comment]
Just noting for QA, the way that this is implemented requires another staging reindex.

**Comment by ShanaLMoore** [comment]
Do you think this is something everyone would want (and should go into iiif_print instead?)

**Comment by laritakr** [comment]
> Do you think this is something everyone would want (and should go into iiif_print instead?)

I think we want to consider doing it like we did child work title, as a lambda, so everyone can configure their own. This implementation would be confusing for organizations that have more than one PDF splitting, since it uses the parent work's title.

File: app/indexers/hyrax/file_set_indexer.rb [hunk 1]
```diff
@@ -62,5 +64,13 @@ def file_format
           object.format_label
         end
       end
+
+      def human_readable_label_name(parent)
+        return unless parent
+
+        work_title = parent.member_of.first&.title&.first
+        page_number = parent.title.first.match(/Page \d+/)&.[](0)
```
Comment (jeremyf) [inline app/indexers/hyrax/file_set_indexer.rb:72]: If we don't have a page number, it seems like we shouldn't check the work_title (which could be expensive)

Consider some short-circuiting logic instead.

**Review by jeremyf — CHANGES_REQUESTED** [review]
See comments

File: app/indexers/hyrax/file_set_indexer.rb [hunk 2]
```diff
@@ -62,5 +64,18 @@ def file_format
           object.format_label
         end
       end
+
+      def human_readable_label_name(parent)
+        return unless parent
+
+        parent_title = parent.title.first
+        page_number = parent_title[/Page \d+/]
```
Comment (jeremyf) [inline app/indexers/hyrax/file_set_indexer.rb:72]: Could you leave a comment as to why this?  Where's this magic regex coming from?  How will future us debug this when we change our title function?

**Review by jeremyf — APPROVED** [review]

**Pull request merged**